File: ios/Zest/ViewModels/MealKitViewModel.swift
import Foundation
import Combine

class MealKitViewModel: ObservableObject {
    @Published var mealKits: [MealKit] = []
    @Published var selectedMealKit: MealKit?
    @Published var isLoading = false
    @Published var errorMessage: String?
    
    private let firebaseService = FirebaseService.shared
    private var cancellables = Set<AnyCancellable>()
    
    init() {
        loadMealKits()
    }
    
    func loadMealKits() {
        isLoading = true
        errorMessage = nil
        Task { @MainActor in
            do {
                self.mealKits = try await firebaseService.getMealKits()
            } catch {
                self.errorMessage = error.localizedDescription
            }
            self.isLoading = false
        }
    }
    
    func loadMealKitDetail(id: String) {
        isLoading = true
        errorMessage = nil
        Task { @MainActor in
            do {
                self.selectedMealKit = try await firebaseService.getMealKit(id: id)
            } catch {
                self.errorMessage = error.localizedDescription
            }
            self.isLoading = false
        }
    }
}

File: android/app/src/main/java/com/neostudios/zest/ui/screens/MealKitViewModel.kt
package com.neostudios.zest.ui.screens

import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.neostudios.zest.domain.model.MealKit
import com.neostudios.zest.domain.repository.MealKitRepository
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.launch
import javax.inject.Inject

@HiltViewModel
class MealKitViewModel @Inject constructor(
    private val mealKitRepository: MealKitRepository
) : ViewModel() {

    private val _uiState = MutableStateFlow(MealKitUiState())
    val uiState: StateFlow<MealKitUiState> = _uiState

    init {
        loadMealKits()
    }

    fun loadMealKits() {
        _uiState.value = _uiState.value.copy(isLoading = true)
        viewModelScope.launch {
            try {
                val mealKits = mealKitRepository.getMealKits()
                _uiState.value = _uiState.value.copy(
                    mealKits = mealKits,
                    isLoading = false
                )
            } catch (e: Exception) {
                _uiState.value = _uiState.value.copy(
                    error = e.localizedMessage,
                    isLoading = false
                )
            }
        }
    }

    fun loadMealKitDetail(id: String) {
        _uiState.value = _uiState.value.copy(isLoading = true)
        viewModelScope.launch {
            try {
                val mealKit = mealKitRepository.getMealKit(id)
                _uiState.value = _uiState.value.copy(
                    selectedMealKit = mealKit,
                    isLoading = false
                )
            } catch (e: Exception) {
                _uiState.value = _uiState.value.copy(
                    error = e.localizedMessage,
                    isLoading = false
                )
            }
        }
    }
}

data class MealKitUiState(
    val mealKits: List<MealKit> = emptyList(),
    val selectedMealKit: MealKit? = null,
    val isLoading: Boolean = false,
    val error: String? = null
)
